internal/controller/gatewayclass_test.go

```diff
@@ -0,0 +1,229 @@
+			}
+
+			// Create
+			Expect(cl.Create(context.Background(), admitted)).Should(Succeed())
+
+			By("Expecting admitted status")
+			Eventually(func() bool {
+				gc := &gatewayv1alpha1.GatewayClass{}
+				_ = cl.Get(context.Background(), admittedKey, gc)
+				return isAdmitted(gc)
+			}, timeout, interval).Should(BeTrue())
+
+			gen, err = generateRandomString(10, charset)
+			Expect(err).NotTo(HaveOccurred())
+
+			notAdmittedKey := types.NamespacedName{Name: "test-gatewayclass-" + gen}
+
+			notAdmitted := &gatewayv1alpha1.GatewayClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      notAdmittedKey.Name,
+					Namespace: notAdmittedKey.Namespace,
+				},
+				Spec: gatewayv1alpha1.GatewayClassSpec{Controller: gcController},
+			}
+
+			// Create
+			Expect(cl.Create(context.Background(), notAdmitted)).Should(Succeed())
+
+			By("Expecting not admitted status")
+			Eventually(func() bool {
+				gc := &gatewayv1alpha1.GatewayClass{}
+				_ = cl.Get(context.Background(), notAdmittedKey, gc)
+				return isAdmitted(gc)
+			}, timeout, interval).Should(BeFalse())
+
+			// Delete admitted gatewayclass
+			By("Expecting to delete successfully")
+			Eventually(func() error {
+				gc := &gatewayv1alpha1.GatewayClass{}
+				_ = cl.Get(context.Background(), admittedKey, gc)
+				return cl.Delete(context.Background(), gc)
+			}, timeout, interval).Should(Succeed())
+
+			By("Expecting to delete finish")
+			Eventually(func() error {
+				gc := &gatewayv1alpha1.GatewayClass{}
+				return cl.Get(context.Background(), admittedKey, gc)
+			}, timeout, interval).ShouldNot(Succeed())
+
+			// Delete non-admitted gatewayclass
+			By("Expecting to delete successfully")
+			Eventually(func() error {
+				gc := &gatewayv1alpha1.GatewayClass{}
+				_ = cl.Get(context.Background(), notAdmittedKey, gc)
+				return cl.Delete(context.Background(), gc)
+			}, timeout, interval).Should(Succeed())
+
+			By("Expecting to delete finish")
+			Eventually(func() error {
+				gc := &gatewayv1alpha1.GatewayClass{}
+				return cl.Get(context.Background(), notAdmittedKey, gc)
+			}, timeout, interval).ShouldNot(Succeed())
+		})
+	})
+	Context("Managed GatewayClass", func() {
+		It("With parameterRefs should not be admitted", func() {
+
+			gen, err := generateRandomString(10, charset)
+			Expect(err).NotTo(HaveOccurred())
+
+			key := types.NamespacedName{Name: "test-gatewayclass-" + gen}
+
+			unsupported := &gatewayv1alpha1.GatewayClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      key.Name,
+					Namespace: key.Namespace,
+				},
+				Spec: gatewayv1alpha1.GatewayClassSpec{
+					Controller: gcController,
+					ParametersRef: &gatewayv1alpha1.ParametersReference{
+						Group: "foo",
+						Kind:  "bar",
+						Name:  "baz",
+					},
+				},
+			}
+
+			// Create
+			Expect(cl.Create(context.Background(), unsupported)).Should(Succeed())
+
+			By("Expecting not admitted status")
+			Eventually(func() bool {
```

Comment:
Should this be a `Consistently` instead? since the default status is admitted==false

Reply:
similar for other places we're creating a gatewayclass and expecting it to not be admitted

Reply:
@sunjayBhatia the reason differs from the [default](https://github.com/kubernetes-sigs/gateway-api/blob/v0.3.0/apis/v1alpha1/gatewayclass_types.go#L44) of "Waiting".

Reply:
doesn't look like `isAdmitted` checks the reason, only looks like it is checking status right now

Reply:
@sunjayBhatia I played around with `Consistently` and had no luck. I found this [ref](https://medium.com/@william.la.martin/ginkgotchas-yeh-also-gomega-13e39185ec96) that states results of a returned func should not be passed to `Consistently`.

> doesn't look like isAdmitted checks the reason, only looks like it is checking status right now

I added `isWaiting` that checks for the "Waiting" status reason.